src/simple_lottery/algorithm.rs

```diff
@@ -0,0 +1,37 @@
+//! Simple lottery prove and verify functions
+
+use super::proof::Proof;
+use super::setup::Setup;
+use super::types::Hash;
+use crate::utils::sample;
+use crate::utils::types::Element;
+use blake2::{Blake2s256, Digest};
+
+pub(super) fn prove(setup: &Setup, prover_set: &[Element]) -> Option<Proof> {
+    let mut element_sequence = Vec::with_capacity(setup.proof_size as usize);
+    for &element in prover_set {
+        if hash(setup, element) {
+            element_sequence.push(element);
+        }
+        if prover_set.len() as u64 >= setup.proof_size {
+            return Some(Proof { element_sequence });
+        }
+    }
+
+    None
+}
+
+pub(super) fn verify(setup: &Setup, proof: &Proof) -> bool {
+    (proof.element_sequence.len() as u64 == setup.proof_size)
+        && proof
+            .element_sequence
+            .iter()
+            .all(|&element| hash(setup, element))
+}
+
+fn hash(setup: &Setup, element: Element) -> bool {
```

Comment:
Perhaps rename this "play_lottery"?

Reply:
I tried to be consistent with `src/centralized_telescope/algorithm.rs`. Let's name it `lottery_hash`?

Reply:
For consistency with the `centralized_telescope` and also expressing the functionality `lottery_hash` is good.

Reply:
done